back/lib/shared/auth/auth.go

```diff
@@ -0,0 +1,80 @@
+package auth
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"math/big"
+
+	"golang.org/x/crypto/scrypt"
+
+	"github.com/KrispyTech/airneis/lib/shared/constants"
+	"github.com/KrispyTech/airneis/lib/shared/helpers"
+)
+
+func CreatePassword(authData helpers.AuthData) (hashedPassword string, salt string, err error) {
+	hashedPassword, salt, err = createNewPasswordHash(authData.Password)
+	if err != nil {
+		return "", "", err
+	}
+
+	return hashedPassword, salt, nil
+}
+
+func createNewPasswordHash(rawPassword string) (password string, salt string, err error) {
+	salt, err = generateSalt()
+	if err != nil {
+		return "", "", err
+	}
+
+	hashedPassword, err := hashPassword([]byte(rawPassword), []byte(salt))
+	if err != nil {
+		return "", "", err
+	}
+
+	password = hex.EncodeToString(hashedPassword)
+
+	return password, salt, nil
+}
+
+func generateSalt() (string, error) {
+	salt := make([]byte, constants.SaltLength)
+	max := big.NewInt(int64(len(constants.Chars)))
+
+	for i := range salt {
+		randomNumber, err := rand.Int(rand.Reader, max)
+		if err != nil {
+			return "", err
+		}
+
+		salt[i] = constants.Chars[randomNumber.Int64()]
+	}
+
+	println(string(salt))
+
+	return string(salt), nil
+}
+
+func hashPassword(rawPassword, salt []byte) ([]byte, error) {
+	const (
+		/*It controls the amount of work required to compute the derived key and directly affects both:
+		CPU time and memory usage.*/
+		memoryCost = 32768
```

Comment:
Would be nice to add some documentation explaining how it works exactly, so that no one changes this value randomly.